packages/jni-swig-stub/realm.i

```diff
@@ -179,63 +210,68 @@ struct realm_size_t {
+    static jmethodID on_change_method = jenv->GetMethodID(notification_class, "onChange", "(J)V");
+
     return realm_object_add_notification_callback(
             object,
             // Use the callback as user data
-            callback,
-            // FIXME MEMORY Verify that JVM callback is actual collected
-            //  https://github.com/realm/realm-kotlin/issues/93
-            [] (void *userdata) {
-                static_cast<NotificationCallback*>(userdata)->~NotificationCallback();
-            },
-            // change callback
-            [] (void* userdata, const realm_object_changes_t* changes) {
-                // FIXME API-NOTIFICATION Consider catching errors and propagate to error callback
-                //  like the C-API error callback below
-                static_cast<NotificationCallback*>(userdata)->onChange(static_cast<const void*>(changes));
-            },
-            // FIXME API-NOTIFICATION Error callback, C-API realm_get_async_error not available yet
-            [] ( void* userdata, const realm_async_error_t* async_error) { },
-            // FIXME NOTIFICATION C-API currently uses the realm's default scheduler
-            NULL
-    );
-}
-realm_notification_token_t* realm_results_add_notification_callbackJNI(
-        realm_results_t* results,
-        NotificationCallback* callback
-) {
-    return realm_results_add_notification_callback(
-            results,
-            // Use the callback as user data
-            callback,
-            // FIXME MEMORY Verify that JVM callback is actual collected
-            //  https://github.com/realm/realm-kotlin/issues/93
-            [] (void *userdata) {
-                static_cast<NotificationCallback*>(userdata)->~NotificationCallback();
-            },
-            // change callback
-            [] (void* userdata, const realm_collection_changes_t* changes) {
-                // FIXME API-NOTIFICATION Consider catching errors and propagate to error callback
-                //  like the C-API error callback below
-                static_cast<NotificationCallback*>(userdata)->onChange(static_cast<const void*>(changes));
-            },
-            // FIXME API-NOTIFICATION Error callback, C-API realm_get_async_error not available yet
-            [] ( void* userdata, const realm_async_error_t* async_error) { },
-            // FIXME NOTIFICATION C-API currently uses the realm's default scheduler
-            NULL
+            static_cast<jobject>(get_env()->NewGlobalRef(callback)),
+    [](void *userdata) {
+        get_env(true)->DeleteGlobalRef(static_cast<jobject>(userdata));
+    },
+    // change callback
+    [](void* userdata, const const realm_object_changes_t* changes) {
+        // FIXME API-NOTIFICATION Consider catching errors and propagate to error callback
+        //  like the C-API error callback below
+        auto jenv = get_env(true);
+        if (jenv->ExceptionCheck()) {
```

Comment:
Same as above...but here you are also throwing an exception??

Reply:
Missed to throw here 👍